lib/toxiproxy/toxic_collection.rb

```diff
@@ -13,34 +13,48 @@ def initialize(proxies)
     end
 
     def apply(&block)
-      @toxics.each(&:enable)
-      yield
-    ensure
-      @toxics.each(&:disable)
+      names = toxics.group_by { |t| [t.name, t.proxy.name] }
+      dups  = names.values.select { |toxics| toxics.length > 1 }
+      if !dups.empty?
+        raise ArgumentError, "There are two toxics with the name #{dups.first[0]} for proxy #{dups.first[1]}, please override the default name (<type>_<direction>)"
```

Comment:
See my comment in https://github.com/Shopify/toxiproxy/pull/83 that we might want to default to `{type}_{stream}`.